app/models/adviser.rb

```diff
@@ -32,6 +32,15 @@ class Adviser < ActiveRecord::Base
   validate :match_reference_number
 
   after_commit :geocode
+  after_save :check_for_changes
+
+  def self.move_to(receiving_firm)
```

Comment:
Can we think of a better name for this? Maybe something both more descriptive of the operation and something that signals its potential destructiveness. Maybe `move_all_to_firm`? I'm thinking along the lines of `destroy_all` and `update_all`.